nengo/spa/basalganglia.py

```diff
@@ -141,3 +143,36 @@ def add_dot_input(self, index, source, symbol, scale):
         with self.spa:
             nengo.Connection(output, self.input[index:index+1],
                              transform=transform, synapse=self.input_synapse)
+
+    def add_scalar_input(self, index, source):
+        """ Add a scalar input that will vary over time.
+
+        This would be used for such a thing as the ouput of the Compare module.
+
+        Parameters
+        ----------
+        index : int
+            the index of the action
+        source : Source
+            the module output to read from
+        symbol : Source
+            the semantic pointer to compute the dot product with
+        scale : float
+            a scaling factor to be applied to the result
+        """
+        output, _ = self.spa.get_module_output(source.name)
+        if(output.size_out != 1):
+            raise NotImplementedError("Only sources with a dimension"
+                                      "of 1 can be scalar inputs")
+
+        try:
+            scale = float(eval(source.transform.symbol))
+        except ValueError:
```

Comment:
How can this ValueError ever occur?  If I try parsing a rule like ```compare*A --> whatever=something``` then I get a ```NameError``` rather than a ```ValueError```.  Should this be NameError instead?  Or is this a Python 2 versus Python 3 thing (I'm on Python 2)?